### PR TITLE
Add storybook

### PR DESCRIPTION
**Pull request by JorikSchellekens**
fixes: https://github.com/matrix-org/matrix.to/issues/98

I've configured story book for all components.
I've configured the accessibility, code, viewport, accessibility and knobs extensions as global decorators.
I've also added a community addon that displays figma designs which is maintained by the developers of figma themselves.

Overall I'm very impressed by this and it's made working on individual components so much faster.

image

### CODE REVIEW

File: .storybook/preview.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { addDecorator } from '@storybook/react';
+import { withA11y } from '@storybook/addon-a11y';
+import { withKnobs } from '@storybook/addon-knobs';
+import { withDesign } from 'storybook-addon-designs'
+import { addParameters } from '@storybook/react';
+
+
+
```
Comment (turt2live) [inline .storybook/preview.js:9]: this feels like excessive whitespace

File: .storybook/main.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+module.exports = {
```
Comment (turt2live) [inline .storybook/main.js:1]: copyright

File: .storybook/preview.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
```
Comment (turt2live) [inline .storybook/preview.js:1]: copyright

File: src/components/CreateLinkTile.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+import CreateLinkTile from "./CreateLinkTile";
+
+export default {
+    title: "CreateLinkTile",
+    parameters: {
+        design: {
+            type: "figma",
+            url:
+                "https://www.figma.com/file/WSXjCGc1k6FVI093qhlzOP/04-Recieving-share-link?node-id=59%3A1",
```
Comment (turt2live) [inline src/components/CreateLinkTile.stories.tsx:27]: I'd just drop `www` from this to make it fit on one line:
```suggestion
            url: "https://figma.com/file/WSXjCGc1k6FVI093qhlzOP/04-Recieving-share-link?node-id=59%3A1",
```

File: src/components/Input.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+import Input from "./Input";
+
+export default {
+    title: "Input",
+    parameters: {
+        design: {
+            type: "figma",
+            url:
+                "https://www.figma.com/file/WSXjCGc1k6FVI093qhlzOP/04-Recieving-share-link?node-id=59%3A1",
```
Comment (turt2live) [inline src/components/Input.stories.tsx:29]:
```suggestion
            url: "https://figma.com/file/WSXjCGc1k6FVI093qhlzOP/04-Recieving-share-link?node-id=59%3A1",
```

File: src/components/TextButton.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+import TextButton from "./TextButton";
+
+export default {
+    title: "TextButton",
+    parameters: {
+        design: {
+            type: "figma",
+            url:
+                "https://www.figma.com/file/WSXjCGc1k6FVI093qhlzOP/04-Recieving-share-link?node-id=149%3A10756",
```
Comment (turt2live) [inline src/components/TextButton.stories.tsx:27]:
```suggestion
            url: "https://figma.com/file/WSXjCGc1k6FVI093qhlzOP/04-Recieving-share-link?node-id=149%3A10756",
```

File: src/components/Tile.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+import Tile from "./Tile";
+
+export default {
+    title: "Tile",
+    parameters: {
+        design: {
+            type: "figma",
+            url:
+                "https://www.figma.com/file/WSXjCGc1k6FVI093qhlzOP/04-Recieving-share-link?node-id=143%3A5853",
```
Comment (turt2live) [inline src/components/Tile.stories.tsx:27]:
```suggestion
            url: "https://figma.com/file/WSXjCGc1k6FVI093qhlzOP/04-Recieving-share-link?node-id=143%3A5853",
```

**Comment by resynth1943** [comment]
Where is the redesign? It looks so good

**Comment by JorikSchellekens** [comment]
Currently the redesign is on [figma](https://www.figma.com/file/WSXjCGc1k6FVI093qhlzOP/04-Recieving-share-link?node-id=59%3A0)! A live version is still a while away.

**Comment by resynth1943** [comment]
Nice!

**Comment by judibwynn** [comment]
Please unsubscribe me!

**Pull request merged**